src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -1551,8 +1541,16 @@ bool AbstractWriteLog<I>::check_allocation(
       m_bytes_allocated += bytes_allocated;
       m_bytes_cached += bytes_cached;
       m_bytes_dirty += bytes_dirtied;
+      if (m_cache_state->clean && bytes_dirtied > 0) {
+        m_cache_state->clean = false;
+        update_image_cache_state();
+        write_image_cache_state(locker);
+      } else {
+        locker.unlock();
```

Comment:
I don't think manual unlocking is necessary here, `unique_lock` destructor  would take care of it.

Reply:
Revised. Thanks, I forgot that it can also be automatically unlocked during deconstruction.